File: External/Program/GM2Calc/examples/example-thdm.cpp
#include "gm2calc/gm2_1loop.hpp"
#include "gm2calc/gm2_2loop.hpp"
#include "gm2calc/gm2_uncertainty.hpp"
#include "gm2calc/gm2_error.hpp"
#include "gm2calc/THDM.hpp"

#include <cstdio>

int main()
{
   gm2calc::thdm::Mass_basis basis;
   basis.yukawa_type = gm2calc::thdm::Yukawa_type::type_2;
   basis.mh = 125;
   basis.mH = 400;
   basis.mA = 420;
   basis.mHp = 440;
   basis.sin_beta_minus_alpha = 0.999;
   basis.lambda_6 = 0;
   basis.lambda_7 = 0;
   basis.tan_beta = 3;
   basis.m122 = 40000;
   basis.zeta_u = 0;
   basis.zeta_d = 0;
   basis.zeta_l = 0;
   basis.Delta_u << 0, 0, 0, 0, 0, 0, 0, 0, 0;
   basis.Delta_d << 0, 0, 0, 0, 0, 0, 0, 0, 0;
   basis.Delta_l << 0, 0, 0, 0, 0, 0, 0, 0, 0;
   basis.Pi_u << 0, 0, 0, 0, 0, 0, 0, 0, 0;
   basis.Pi_d << 0, 0, 0, 0, 0, 0, 0, 0, 0;
   basis.Pi_l << 0, 0, 0, 0, 0, 0, 0, 0, 0;

   gm2calc::SM sm;
   sm.set_alpha_em_mz(1.0/128.94579);
   sm.set_mu(2, 173.34);
   sm.set_mu(1, 1.28);
   sm.set_md(2, 4.18);
   sm.set_ml(2, 1.77684);

   gm2calc::thdm::Config config;
   config.force_output = false;
   config.running_couplings = true;

   try {
      const gm2calc::THDM model(basis, sm, config);

      const double amu = gm2calc::calculate_amu_1loop(model)
                       + gm2calc::calculate_amu_2loop(model);

      const double delta_amu =
         gm2calc::calculate_uncertainty_amu_2loop(model);

      std::printf("amu = %.5e +- %.5e\n", amu, delta_amu);
   } catch (const gm2calc::Error& e) {
      std::printf("%s\n", e.what());
   }

   return 0;
}

File: External/Program/GM2Calc/examples/example-thdm.m
Install["bin/gm2calc.mx"];

GM2CalcSetFlags[
    loopOrder -> 2,
    tanBetaResummation -> True,
    forceOutput -> False,
    runningCouplings -> True];

GM2CalcSetSMParameters[
    alpha0 -> 0.00729735,
    alphaMZ -> 0.0077552,
    alphaS -> 0.1184,
    MhSM -> 125.09,
    MW -> 80.385,
    MZ -> 91.1876,
    MT -> 173.34,
    mcmc -> 1.28,
    mu2GeV -> 0.0022,
    mbmb -> 4.18,
    ms2GeV -> 0.096,
    md2GeV -> 0.0047,
    ML -> 1.777,
    MM -> 0.1056583715,
    ME -> 0.000510998928,
    Mv1 -> 0,
    Mv2 -> 0,
    Mv3 -> 0,
    CKM -> IdentityMatrix[3] ];

(* calculate amu using the gauge basis input parameters *)
Print[{amu, Damu} /. GM2CalcAmuTHDMGaugeBasis[
    yukawaType        -> 2,
    lambda            -> { 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1 },
    TB                -> 3,
    m122              -> 200^2,
    zetau             -> 0,
    zetad             -> 0,
    zetal             -> 0,
    Deltau            -> 0 IdentityMatrix[3],
    Deltad            -> 0 IdentityMatrix[3],
    Deltal            -> 0 IdentityMatrix[3],
    Piu               -> 0 IdentityMatrix[3],
    Pid               -> 0 IdentityMatrix[3],
    Pil               -> 0 IdentityMatrix[3]]
];

(* calculate amu using the mass basis input parameters *)
Print[{amu, Damu} /. GM2CalcAmuTHDMMassBasis[
    yukawaType        -> 2,
    Mhh               -> { 125, 400 },
    MAh               -> 420,
    MHp               -> 440,
    sinBetaMinusAlpha -> 0.999,
    lambda6           -> 0,
    lambda7           -> 0,
    TB                -> 3,
    m122              -> 200^2,
    zetau             -> 0,
    zetad             -> 0,
    zetal             -> 0,
    Deltau            -> 0 IdentityMatrix[3],
    Deltad            -> 0 IdentityMatrix[3],
    Deltal            -> 0 IdentityMatrix[3],
    Piu               -> 0 IdentityMatrix[3],
    Pid               -> 0 IdentityMatrix[3],
    Pil               -> 0 IdentityMatrix[3]]
];
